src/languages/php.js

```diff
@@ -374,7 +374,8 @@ export default function(hljs) {
       FUNCTION_INVOKE,
       {
         // swallow composed identifiers to avoid parsing them as keywords
-        begin: /(::|->)+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/
+        begin: /(::|->)+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*(?! *\()(?![a-zA-Z0-9_\x7f-\xff])/,
```

Comment:
> _I'm not sure they all needed to change_, unless the `[ ]` is more precise? (it may likely be) IIRC the issue was with this line alone because `\xA0*` overlaps with `\s*`.... which is fixed easily enough by changing the look-ahead \s to a space, if that works for our use cases. :-) The character groups in the regex here also could have been fixed. (I might just do this)
> 
> So, is this more precise? (in which case I'll likely change them to `[ ]` for readability.
> 
> Any final thoughts? Thanks for all the hard work on this. I think the finish line is just ahead. :)

```
new    
\DateTime   
();
```
This is also allowed so I will introduce const SEPARATOR `[ \t\n]`. This should cover all the cases. Probably this will be better in all those places, if non-breaking space may be part of the name and `\s` select it.

Reply:
> This is also allowed 

I imagined it might be. :-)

> so I will introduce const SEPARATOR

I think "WHITESPACE" might be better (and add a comment saying it purposely excludes `0xA0`), etc...

Reply:
> I think "WHITESPACE" might be better (and add a comment saying it purposely excludes `0xA0`), etc...

Done @joshgoebel